include/dxc/Support/HLSLOptions.td

```diff
@@ -229,6 +229,12 @@ def _version : Flag<["--"], "version">, Group<hlslcore_Group>, Flags<[DriverOpti
 
 def ast_dump : Flag<["-", "/"], "ast-dump">, Flags<[CoreOption, DriverOption, HelpHidden]>,
   HelpText<"Dumps the parsed Abstract Syntax Tree.">; // should not be core, but handy workaround until explicit API written
+def dump_dependencies : Flag<["-", "/"], "M">, Flags<[CoreOption, DriverOption]>,
+  HelpText<"Dumps the list of the compilation target dependencies.">;
+def write_dependencies : Flag<["-", "/"], "MD">, Flags<[CoreOption, DriverOption]>,
+  HelpText<"Write a file with .d extension that will contain the list of the compilation target dependencies.">;
+def write_dependencies_to : JoinedOrSeparate<["-", "/"], "MF">, MetaVarName<"<file>">, Flags<[CoreOption, DriverOption]>,
```

Comment:
Both `write_dependencies` and `write_dependencies_to` appear to only be implemented by dxc.exe, so I don't think they should include `CoreOption` here.

Also, just as an aside about having these at all: we don't have these sorts of extra options for automatically writing to a file based on the input filename or specifying an output filename specifically for other analogous options (like ast-dump), so while this is consistent with clang's driver options, it feels a bit like option bloat.  It's not hard to redirect stdout.  But that's just a minor quibble, and we should probably keep these for convenience if that's what people want - just limited to `DriverOption` since they are only in dxc.exe, not the DxCompiler API.

Reply:
Done.

> Also, just as an aside about having these at all: we don't have these sorts of extra options for automatically writing to a file based on the input filename or specifying an output filename specifically for other analogous options (like ast-dump), so while this is consistent with clang's driver options, it feels a bit like option bloat. It's not hard to redirect stdout. But that's just a minor quibble, and we should probably keep these for convenience if that's what people want - just limited to DriverOption since they are only in dxc.exe, not the DxCompiler API.

I do not have a strong opinion on it. As far as I remember, `-MF` and `-MD` was suggested by Chris in our meeting. I agreed on it because `-MD` is what CMake uses to keep/check the dependencies.